src/lib.rs

```diff
@@ -44,20 +44,25 @@ use std::fs::File;
 use std::io::prelude::*;
 use std::net::{TcpListener, TcpStream};
 use std::path::Path;
+use std::time::Duration;
 
 mod error;
+mod request;
 
 pub use error::Error;
 
+
+type ServerHandler = fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>;
+
 /// A web server.
 ///
 /// This is the core type of this crate, and is used to create a new
 /// server and listen for connections.
 pub struct Server {
-    handler: fn(Request<&[u8]>, ResponseBuilder) -> Result<Response<&[u8]>, Error>,
+    handler: ServerHandler,
+    timeout: Option<Duration>,
```

Comment:
it's unfortunate that https://github.com/steveklabnik/simple-server/issues/64 hasn't borne fruit yet; futures make this kind of stuff super easy and don't require specific coding. That said, I don't think that this should block this or anything, just wanted to say "darn" 😄